pkg/local_workflows/output_workflow.go

```diff
@@ -36,13 +39,50 @@ func InitOutputWorkflow(engine workflow.Engine) error {
 	outputConfig.String(OUTPUT_CONFIG_KEY_SARIF_FILE, "", "Write sarif output to file")
 	outputConfig.Bool(configuration.FLAG_INCLUDE_IGNORES, false, "Include ignored findings in the output")
 	outputConfig.Bool(configuration.FLAG_ONLY_IGNORES, false, "Hide open issues in the output")
+	outputConfig.String(configuration.FLAG_SEVERITY_THRESHOLD, "low", "Severity threshold for findings to be included in the output")
 
 	entry, err := engine.Register(WORKFLOWID_OUTPUT_WORKFLOW, workflow.ConfigurationOptionsFromFlagset(outputConfig), outputWorkflowEntryPointImpl)
 	entry.SetVisibility(false)
 
 	return err
 }
 
+func filterSummaryOutput(config configuration.Configuration, input workflow.Data) (workflow.Data, error) {
+	// Parse the summary data
+	summary := json_schemas.NewTestSummary("")
+	payload, ok := input.GetPayload().([]byte)
+	if !ok {
+		return nil, fmt.Errorf("invalid payload type: %T", input.GetPayload())
+	}
+	err := json.Unmarshal(payload, &summary)
+	if err != nil {
+		return input, err
+	}
+
+	minSeverity := config.GetString(configuration.FLAG_SEVERITY_THRESHOLD)
+	minSeverityLevel := slices.Index(summary.SeverityOrderAsc, minSeverity)
```

Comment:
Suggestion: we currently have the same filter logic in multiple places, which makes it easy to diverge. How about changing the filtering concept a bit and thereby enable to re-use some of the logic.

1. Apply severity threshold filter on slice SeverityOrderAsc, the resulting slice contains only the Severity Levels that are above the threshold.
2. Use the FilteredSeverityOrderAsc to iterate over TestSummary
3. Use the FilteredSeverityOrderAsc to iterate over Findings

Reply:
applied